extensions/skyportal/static/js/components/Filter.jsx

```diff
@@ -0,0 +1,401 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Link, useParams } from "react-router-dom";
+
+import Paper from "@material-ui/core/Paper";
+import { makeStyles, useTheme } from "@material-ui/core/styles";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Switch from "@material-ui/core/Switch";
+import FormHelperText from "@material-ui/core/FormHelperText";
+import FormControl from "@material-ui/core/FormControl";
+import Select from "@material-ui/core/Select";
+import InputLabel from "@material-ui/core/InputLabel";
+import MenuItem from "@material-ui/core/MenuItem";
+import Grid from "@material-ui/core/Grid";
+import Accordion from "@material-ui/core/Accordion";
+import AccordionSummary from "@material-ui/core/AccordionSummary";
+import AccordionDetails from "@material-ui/core/AccordionDetails";
+import Typography from "@material-ui/core/Typography";
+import TextareaAutosize from "@material-ui/core/TextareaAutosize";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+import Button from "@material-ui/core/Button";
+import Card from "@material-ui/core/Card";
+import CardActions from "@material-ui/core/CardActions";
+import CardContent from "@material-ui/core/CardContent";
+import CircularProgress from "@material-ui/core/CircularProgress";
+
+import ReactDiffViewer from "react-diff-viewer";
+import { useForm } from "react-hook-form";
+import { showNotification } from "baselayer/components/Notifications";
+
+import * as groupActions from "../ducks/group";
+import * as filterActions from "../ducks/filter";
+import * as filterVersionActions from "../ducks/kowalski_filter";
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    width: "100%",
+    padding: theme.spacing(1),
+    textAlign: "left",
+    color: theme.palette.text.primary,
+  },
+  nested: {
+    paddingLeft: theme.spacing(1),
+  },
+  heading: {
+    fontSize: "1.0625rem",
+    fontWeight: 500,
+  },
+  accordion_details: {
+    flexDirection: "column",
+  },
+  button_add: {
+    maxWidth: "8.75rem",
+  },
+  formControl: {
+    margin: theme.spacing(1),
+    minWidth: "12rem",
+  },
+  selectEmpty: {
+    marginTop: theme.spacing(2),
+  },
+  root: {
+    minWidth: "18rem",
+  },
+  bullet: {
+    display: "inline-block",
+    margin: "0 2px",
+    transform: "scale(0.8)",
+  },
+  title: {
+    fontSize: "0.875rem",
+  },
+  big_font: {
+    fontSize: "1rem",
+  },
+  pos: {
+    marginBottom: "0.75rem",
+  },
+}));
+
+const Filter = () => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const { register, handleSubmit } = useForm();
+
+  const [filterLoadError, setFilterLoadError] = useState("");
+  const [filterVersionLoadError, setFilterVersionLoadError] = useState("");
+  const [groupLoadError, setGroupLoadError] = useState("");
```

Comment:
filterVersionLoadError and groupLoadError gets set on errors but never used anywhere - maybe show them as notifications?

Reply:
Done